libraries/chain/database.cpp

```diff
@@ -2283,12 +2286,49 @@ namespace golos { namespace chain {
                             unclaimed_rewards -= claim;
 
                             const auto &voter = get(itr->voter);
-                            auto reward = create_vesting(voter, asset(claim, STEEM_SYMBOL));
+
+                            auto voter_reward = claim;
+
+                            if (has_hardfork(STEEMIT_HARDFORK_0_19__756)) {
+                                for (auto& dvir : itr->delegator_vote_interest_rates) {
+                                    auto delegator_claim = claim * dvir.interest_rate / STEEMIT_100_PERCENT;
+
+                                    if (delegator_claim == 0) {
+                                        continue;
+                                    }
+
+                                    voter_reward -= delegator_claim;
+
+                                    const auto& delegator = get_account(dvir.account);
+                                    asset delegator_vesting = create_vesting(delegator, asset(delegator_claim, STEEM_SYMBOL));
+                                    modify(delegator, [&](account_object& a) {
+                                        a.delegated_vesting_shares += delegator_vesting;
+                                    });
+                                    modify(voter, [&](account_object& a) {
+                                        a.received_vesting_shares += delegator_vesting;
+                                    });
+
+                                    auto vdo_itr = vdo_idx.lower_bound(voter.name);
+                                    if (vdo_itr != vdo_idx.end() && vdo_itr->delegatee == voter.name) {
+                                        modify(*vdo_itr, [&](vesting_delegation_object& o) {
+                                            o.vesting_shares += delegator_vesting;
+                                        });
+                                        auto vdeo_itr = vdeo_idx.find(std::make_tuple(vdo_itr->delegatee, vdo_itr->delegator));
+                                        for (; vdeo_itr != vdeo_idx.end(); ++vdeo_itr) {
+                                            modify(*vdeo_itr, [&](vesting_delegation_expiration_object& o) {
+                                                o.vesting_shares += delegator_vesting;
+                                            });
+                                        }
```

Comment:
what about else case?

Reply:
@afalaleev Yes, you are right. VDEO could be without VDO. Because you can remove VDO with operation but it don't removing VDEO - they removing only when expired.